test/steps/kogitoapp.go

```diff
@@ -68,77 +68,90 @@ func (data *Data) deployQuarkusExampleServiceWithNativeAndLabels(contextDir, nat
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.QuarkusRuntimeType,
-			Native:      framework.MustParseEnabledDisabled(native),
-			Persistence: true,
-			Events:      true,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.QuarkusRuntimeType,
+			Native:             framework.MustParseEnabledDisabled(native),
+			Persistence:        true,
+			Events:             true,
+			Labels:             nil,
 		})
 }
 
 func (data *Data) deploySpringBootExampleService(contextDir string) error {
-	return framework.DeployExample(data.Namespace, framework.GetDefaultInstallerType(),
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.SpringbootRuntimeType,
-			Native:      false,
-			Persistence: false,
-			Events:      false,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.SpringbootRuntimeType,
+			Native:             false,
+			Persistence:        false,
+			Events:             false,
+			Labels:             nil,
 		})
 }
 
 func (data *Data) deploySpringBootExampleServiceWithPersistence(contextDir string) error {
-	return framework.DeployExample(data.Namespace, framework.GetDefaultInstallerType(),
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
 		framework.KogitoAppDeployment{
-			AppName:     filepath.Base(contextDir),
-			ContextDir:  contextDir,
-			Runtime:     v1alpha1.SpringbootRuntimeType,
-			Native:      false,
-			Persistence: true,
-			Events:      false,
-			Labels:      nil,
+			AppName:            filepath.Base(contextDir),
+			GitSourceURI:       config.GetExamplesRepositoryURI(),
+			GitSourceReference: config.GetExamplesRepositoryRef(),
+			ContextDir:         contextDir,
+			Runtime:            v1alpha1.SpringbootRuntimeType,
+			Native:             false,
+			Persistence:        true,
+			Events:             false,
+			Labels:             nil,
 		})
 }
 
-func (data *Data) deployServiceFromExampleFile(exampleFile string) error {
-	return framework.DeployServiceFromExampleFile(data.Namespace, exampleFile)
+func (data *Data) createService(serviceName string) error {
+	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(),
+		framework.KogitoAppDeployment{
+			AppName: serviceName,
+			Runtime: v1alpha1.SpringbootRuntimeType,
+		})
```

Comment:
Just if I get the binary build right...This will just deploy the custom resource of the service without building it as it misses the git URL etc.? After that the operator will create a BuildConfig resource which will be bound with the actual start of build by the build name? Thanks!

Reply:
yes, exactly.